Aliucord/src/main/java/com/aliucord/coreplugins/slashcommandsfix/Patches.java

```diff
@@ -0,0 +1,315 @@
+
+        var applications = new ArrayList<Application>(applicationIndex.applications.values());
+        Collections.sort(applications, (left, right) -> left.getName().compareTo(right.getName()));
+        applications.add(((BuiltInCommandsProvider) this.storeApplicationCommandsBuiltInCommandsProviderField.get(storeApplicationCommands)).getBuiltInApplication());
+        this.handleGuildApplicationsUpdateMethod.invoke(storeApplicationCommands, applications);
+
+        switch (requestSource) {
+            case BROWSE:
+                this.handleDiscoverCommandsUpdateMethod.invoke(storeApplicationCommands, new ArrayList(applicationIndex.applicationCommands.values()));
+                break;
+
+            case QUERY:
+                this.handleQueryCommandsUpdateMethod.invoke(storeApplicationCommands, new ArrayList(applicationIndex.applicationCommands.values()));
+                break;
+        }
+    }
+
+    private ApplicationIndex requestApplicationIndex(ApplicationIndexSource source) {
+        // Reuse application index from cache
+        var applicationIndex = source.getFromCache(applicationIndexCache);
+        if (!applicationIndex.isPresent()) {
+            try {
+                // Request application index from API
+                applicationIndex = Optional.of(
+                    Http.Request.newDiscordRNRequest(source.getEndpoint())
+                        .execute()
+                        .json(GsonUtils.getGsonRestApi(), ApiApplicationIndex.class)
+                        .toModel()
+                );
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+
+            source.insertIntoCache(applicationIndexCache, applicationIndex.get());
+        }
+        return applicationIndex.get();
+    }
+
+    private void cleanApplicationIndexCache(ApplicationIndexSource source) {
+        source.removeFromCache(applicationIndexCache);
+    }
+
+    private static Optional<ApplicationIndexSource> applicationIndexSourceFromContext(long guildId, StoreChannelsSelected storeChannelsSelected) {
+        Optional<ApplicationIndexSource> applicationIndexSource = Optional.empty();
+        // guildId being 0 means this is a DM or a DM group
+        if (guildId != 0) {
+            applicationIndexSource = Optional.of(new ApplicationIndexSourceGuild(guildId));
+        } else {
+            // Only create a DM index source for bots
+            var channel = storeChannelsSelected.getSelectedChannel();
+            var channelType = channel.D();
+            if (channelType == Channel.DM) {
+                var user = channel.z().get(0);
+                var userIsBot = Optional.ofNullable(user.e())
+                    .orElse(false);
+                if (userIsBot) {
+                    var channelId = channel.k();
+                    applicationIndexSource = Optional.of(new ApplicationIndexSourceDm(channelId));
+                }
```

Comment:
Could you use the UserWrapper/ChannelWrapper classes here instead of directly calling the obfuscated methods